src/blocks/CarouselBlock.php

```diff
@@ -81,13 +83,11 @@ public function images()
         foreach ($this->getVarValue('images', []) as $item) {
             $image = BlockHelper::imageUpload($item['image'], false, true);
             if ($image) {
-                if (isset($item['caption']) && !empty($item['caption'])) {
-                    $image->caption = $item['caption'];
-                }
                 $images[] = [
                     'image' => $image,
-                    'title' => isset($item['title']) ?: null,
-                    'link' => isset($item['link']) ? BlockHelper::linkObject($item['link']) : null,
+                    'title' => isset($item['title']) ? $item['title'] : null,
+                    'caption' => isset($item['caption']) ? $item['caption'] : null,
+                    'link' => (isset($item['link']) && !empty(BlockHelper::linkObject($item['link']))) ? BlockHelper::linkObject($item['link']) : null,
```

Comment:
The link object only requires the isset $item['link'] - why the other checks are required? see what linkObject does: https://github.com/luyadev/luya-module-cms/blob/master/src/helpers/BlockHelper.php#L265-L278

Reply:
Hm.. Good question.. No idea what I did there..